bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/StackRenderer.java

```diff
@@ -623,6 +667,38 @@ public Object createWidget(MUIElement element, Object parent) {
 		return tabFolder;
 	}
 
+	/**
+	 * @param tabFolder
+	 */
+	private void createOnboardingControls(OnboardingCTabFolder tabFolder) {
+		Composite container = WidgetFactory.composite(SWT.None).layout(GridLayoutFactory.swtDefaults().create())
+				.create(tabFolder);
+
+		GridDataFactory gridData = GridDataFactory
+				.create(GridData.VERTICAL_ALIGN_CENTER | GridData.HORIZONTAL_ALIGN_CENTER).grab(true, true);
+
+		Composite content = WidgetFactory.composite(SWT.NONE).supplyLayoutData(gridData::create)
+				.layout(GridLayoutFactory.swtDefaults().create()).create(container);
+
+		Color color = container.getDisplay().getSystemColor(SWT.COLOR_WIDGET_DISABLED_FOREGROUND);
+		welcomeText = WidgetFactory.label(SWT.NONE).foreground(color).supplyLayoutData(gridData::create)
+				.create(content);
+
+		welcomeImage = WidgetFactory.label(SWT.NONE).foreground(color).supplyLayoutData(gridData::create)
+				.create(content);
+
+		tabFolder.setOnboarding(container);
+
+		tabFolder.addDisposeListener(e -> {
+			if (!color.isDisposed()) {
+				color.dispose();
+			}
+			if (welcomeImage.getImage() != null && !welcomeImage.getImage().isDisposed()) {
+				welcomeImage.dispose();
```

Comment:
This looks strange, we've created two images but dispose one. Easier is to add dispose listener to the control on which you supply images at the moment you create them.

Reply:
Done! But I dont really like it. I keep the onboarding as state now.
together with welcomeImage and welcomeText this is a lot new state. I'm not a fan of state.